upgrade/upgrade-scripts/upgrade

```diff
@@ -395,16 +397,21 @@ function finalize() {
 		;;
 	esac
 
-	ROOTFS_DATASET=$(get_mounted_rootfs_container_dataset)
-	[[ -n "$ROOTFS_DATASET" ]] ||
-		die "unable to determine mounted rootfs container dataset"
-
-	SNAPSHOT_NAME=$(get_dataset_rollback_snapshot_name "$ROOTFS_DATASET")
-	[[ -n "$SNAPSHOT_NAME" ]] ||
-		die "unable to determine rollback snapshot name"
-
-	zfs destroy -r "$ROOTFS_DATASET@$SNAPSHOT_NAME" ||
-		die "failed to destroy rollback snapshots"
+	SNAPSHOT_LIST=$(zfs list -t snapshot -r rpool/ROOT -d 2 -Ho name) ||
```

Comment:
We're not filtering the list of snapshots to _only_ the snapshot names that we expect (i.e. `execute-upgrade.XXXXXXX`). I think this is fine, since I don't believe we take any other snapshots on these datasets, but I just wanted to make a note of it.

Additionally, I think not having any filtering logic is preferable for now, since we wouldn't have any way to test the filtering (since we don't have any other snapshot names that'd we want to filter out).

Reply:
That makes sense, I'll keep that in mind for future developments.